packages/tool-cache/src/tool-cache.ts

```diff
@@ -526,13 +526,15 @@ export async function getManifestFromRepo(
 export async function findFromManifest(
   versionSpec: string,
   stable: boolean,
-  manifest: IToolRelease[]
+  manifest: IToolRelease[],
+  archFilter: string = os.arch()
```

Comment:
I implemented the easiest solution and just added one more parameter. But may be it makes sense to use some kind of object here, like :
```
interface IManifestMatchFilter {
  version: string;
  stable:? boolean;
  archFilter:? string;
  platFilter:? string;
}

export async function findFromManifest(
  manifest: IToolRelease[],
  filter: IManifestMatchFilter;
}
```
It will allow us to extend this interface in future if we need. If it sounds good to you, I can rework this pull-request.

If you prefer to keep current parameters, I would suggest to change order of parameters:
```
manifest,
versionSpec,
stable,
archFilter
```
It will be more logical

Reply:
I think another param is fine for now.  If we ever do a v2 where we break compat, we can take a pass and put all options in options objs.